core/tests/flow.rs

```diff
@@ -2,336 +2,219 @@
-//     let anyone_can_spend_amount = script_builder::anyone_can_spend_txout().value;
-//     let expected_withdraw_amount = Amount::from_sat(BRIDGE_AMOUNT_SATS - 2 * config.min_relay_fee)
-//         - anyone_can_spend_amount * 2;
-//     assert_eq!(expected_withdraw_amount, rpc_withdraw_amount);
-// }
+use bitcoin::Address;
+use clementine_circuits::constants::BRIDGE_AMOUNT_SATS;
+use clementine_core::config::BridgeConfig;
+use clementine_core::database::common::Database;
+use clementine_core::errors::BridgeError;
+use clementine_core::extended_rpc::ExtendedRpc;
+use clementine_core::mock::common;
+use clementine_core::musig2::aggregate_nonces;
+use clementine_core::servers::*;
+use clementine_core::traits::rpc::OperatorRpcClient;
+use clementine_core::traits::rpc::VerifierRpcClient;
+use clementine_core::user::User;
+use clementine_core::EVMAddress;
+use clementine_core::UTXO;
+use clementine_core::{
+    create_extended_rpc, create_test_config, create_test_config_with_thread_name,
+};
+use std::thread;
+
+#[tokio::test]
+async fn test_deposit() -> Result<(), BridgeError> {
+    let mut config = create_test_config_with_thread_name!("test_config_flow_1.toml");
+    let rpc = create_extended_rpc!(config);
+
+    // Create temporary databases for testing.
+    let handle = thread::current()
+        .name()
+        .unwrap()
+        .split(':')
+        .last()
+        .unwrap()
+        .to_owned();
+    for i in 0..4 {
+        create_test_config!(
+            handle.clone() + i.to_string().as_str(),
+            "test_config_flow_1.toml"
+        );
+    }
+
+    let mut operators = Vec::new();
+    let mut evm_addresses = Vec::new();
+    for i in 0..config.num_operators {
+        let secret_key = config.all_operators_secret_keys.clone().unwrap()[i].clone();
+        let operator = create_operator_server(
+            BridgeConfig {
+                secret_key,
+                ..config.clone()
+            },
+            rpc.clone(),
+        )
+        .await
+        .unwrap();
+
+        operators.push((operator.0, operator.1, secret_key));
+        evm_addresses.push(EVMAddress([(i + 1) as u8; 20]));
+    }
+    let mut verifiers = Vec::new();
+    for i in 0..config.num_verifiers {
+        verifiers.push(
+            create_verifier_server(
+                BridgeConfig {
+                    secret_key: config.all_verifiers_secret_keys.clone().unwrap()[i].clone(),
+                    ..config.clone()
+                },
+                rpc.clone(),
+            )
+            .await
+            .unwrap(),
+        );
+    }
+
+    println!("Operators: {:#?}", operators);
+    println!("Verifiers: {:#?}", verifiers);
+
+    let secret_key = secp256k1::SecretKey::new(&mut secp256k1::rand::thread_rng());
+
+    let user = User::new(
+        rpc.clone(),
+        config.verifiers_public_keys.clone(),
+        secret_key,
+        config.clone(),
+    );
+
+    let deposit_outpoints: Vec<_> = evm_addresses
```

Comment:
We should generate a single deposit_outpoint for this test, we will just test the flow of a single deposit